src/OutputWriters/checkpointer.jl

```diff
@@ -93,23 +93,28 @@ defaultname(::Checkpointer, nelems) = :checkpointer
 convert_to_arch(::CPU, a) = a
 convert_to_arch(::GPU, a) = CuArray(a)
 
-function restore_if_not_missing(file, location)
-    if haskey(file, location)
-        return file[location]
+function restore_if_not_missing(file, address)
+    if haskey(file, address)
+        return file[address]
     else
-        @warn "Checkpoint file does not contain $location. Returning missing. " *
+        @warn "Checkpoint file does not contain $address. Returning missing. " *
               "You might need to restore this manually."
         return missing
     end
 end
 
-function restore_field(file, location, arch, grid)
-    field_location = file[location * "/location"]
-    data = OffsetArray(convert_to_arch(arch, file[location * "/data"]), grid)
-    bcs = restore_if_not_missing(file, location * "/boundary_conditions")
-    return Field(field_location, arch, grid, bcs, data)
+function restore_field(file, address, arch, grid, loc)
+    field_address = file[address * "/address"]
+    data = OffsetArray(convert_to_arch(arch, file[address * "/data"]), grid, loc)
+    bcs = restore_if_not_missing(file, address * "/boundary_conditions")
+    return Field(field_address, arch, grid, bcs, data)
 end
```

Comment:
This seems wrong. Maybe it's leftover from a search and replace but for every field we checkpoint there is, e.g.
```julia
file["velocities/u/location"]
file["velocities/u/data"]
file["velocities/u/boundary_conditions"]
```
I'll make suggestions that should fix this.

This is partly my fault for mixing field location with JLD2 file location... Good idea on differentiation between `address` and `location`.

Reply:
Ah ok, perhaps I did screw up search and replace.